src/Libraries/Microsoft.Extensions.DataIngestion/Processors/SummaryEnricher.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.AI;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.DataIngestion;
+
+/// <summary>
+/// Enriches chunks with summary text using an AI chat model.
+/// </summary>
+/// <remarks>
+/// It adds "summary" text metadata to each chunk.
+/// </remarks>
+public sealed class SummaryEnricher : IngestionChunkProcessor<string>
+{
+    private readonly IChatClient _chatClient;
+    private readonly ChatOptions? _chatOptions;
+    private readonly TextContent _request;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SummaryEnricher"/> class.
+    /// </summary>
+    /// <param name="chatClient">The chat client used for summary generation.</param>
+    /// <param name="chatOptions">Options for the chat client.</param>
+    /// <param name="maxWordCount">The maximum number of words for the summary. When not provided, it defaults to 100.</param>
+    public SummaryEnricher(IChatClient chatClient, ChatOptions? chatOptions = null, int? maxWordCount = null)
+    {
+        _chatClient = Throw.IfNull(chatClient);
+        _chatOptions = chatOptions;
+
+        int wordCount = maxWordCount.HasValue ? Throw.IfLessThanOrEqual(maxWordCount.Value, 0, nameof(maxWordCount)) : 100;
+        _request = new($"Write a summary text for this text with less than {wordCount} words. Return just the summary.");
```

Comment:
What if wordCount is 1?

Reply:
I don't want to enforce any min value, as it would need to be just a particular number and could open an endless discussion of "why X and not Y?".